### PR TITLE
Basic types/values entries for users guide

### PR DESCRIPTION
**Pull request by bradcray**
This introduces proposed pages for basic types/values (bools, ints/uints, reals/imags/complexes, strings) to the user's guide along with supporting tests.

### CODE REVIEW

File: doc/sphinx/source/users-guide/base/basicTypes.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+:title: Users Guide: Basic Types
+
+Basic Types: Booleans, Numbers, and Strings
+===========================================
+
+Chapel supports a variety of basic types to represent logical values,
+numbers, and strings.  Specifically, these types provide support for:
+Boolean values; signed and unsigned integers; real, imaginary, and
```
Comment (psahabu) [inline doc/sphinx/source/users-guide/base/basicTypes.rst]: I'd consider making this a bulleted list.

File: test/release/examples/users-guide/base/floatValues.chpl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+writeln(myReal);
+myReal = 123.0e2;
+writeln(myReal);
+myReal = 123e2;
+writeln(myReal);
+myReal = 123.0E2;
+writeln(myReal);
+myReal = 123E2;
+writeln(myReal);
+//myReal = 0x1ff;
```
Comment (psahabu) [inline test/release/examples/users-guide/base/floatValues.chpl]: basicValues says this should work, so why is this commented out?

**Comment by psahabu** [comment]
Sorry for the delay. Other than my two comments above, looks good to me.

**Comment by bradcray** [comment]
@psahabu : I switched to the bulleted list you suggested and figured out how to get hex float values working.  Also figured out how to cross-link from the second document back to the first.  And did some wordsmithing as well while here.  Let me know if you want to take another look or not.

**Comment by psahabu** [comment]
Took a brief look, you should be good.

**Pull request merged**